kernel/packages/shared/loading/ReportFatalError.ts

```diff
@@ -54,47 +61,83 @@ export function bringDownClientAndShowError(event: ExecutionLifecycleEvent) {
   aborted = true
 }
 
-export type FatalErrorInfo = {
-  type: string
-  message: string
-  stack?: string
-  sagaStack?: string
-  filename?: string
+export namespace ErrorContext {
+  export const WEBSITE_INIT = `website#init`
+  export const COMMS_INIT = `comms#init`
+  export const KERNEL_INIT = `kernel#init`
+  export const KERNEL_SAGA = `kernel#saga`
+  export const KERNEL_SCENE = `kernel#scene`
+  export const RENDERER_AVATARS = `renderer#avatars`
+  export const RENDERER_ERRORHANDLER = `renderer#errorHandler`
 }
 
-export function ReportFatalError(event: ExecutionLifecycleEvent, errorInfo?: FatalErrorInfo) {
-  bringDownClientAndShowError(event)
-  if (ExecutionLifecycleEventsList.includes(event)) {
-    return globalThis.globalStore && globalThis.globalStore.dispatch(action(event))
+export type ErrorContextTypes =
+  | typeof ErrorContext.WEBSITE_INIT
+  | typeof ErrorContext.COMMS_INIT
+  | typeof ErrorContext.KERNEL_INIT
+  | typeof ErrorContext.KERNEL_SAGA
+  | typeof ErrorContext.KERNEL_SCENE
+  | typeof ErrorContext.RENDERER_AVATARS
+  | typeof ErrorContext.RENDERER_ERRORHANDLER
+
+export function ReportFatalErrorWithCatalystPayload(error: Error, context: ErrorContextTypes) {
+  // TODO(Brian): Get some useful catalyst payload to append here
+  ReportFatalError(error, context)
+}
+
+export function ReportFatalErrorWithCommsPayload(error: Error, context: ErrorContextTypes) {
+  // TODO(Brian): Get some useful comms payload to append here
+  ReportFatalError(error, context)
+}
+
+export function ReportFatalErrorWithUnityPayload(error: Error, context: ErrorContextTypes) {
+  ReportFatalErrorWithUnityPayloadAsync(error, context)
+    .then((x) => {
+      //
+    })
+    .catch(() => {
+      //
+    })
+}
+
+export async function ReportFatalErrorWithUnityPayloadAsync(error: Error, context: ErrorContextTypes) {
+  try {
+    let payload = await unityInterface.CrashPayloadRequest()
+    ReportFatalError(error, context, { "rendererPayload": payload } )
+  } catch (e) {
+    ReportFatalError(error, context)
   }
-  trackEvent('generic_error', {
-    message: event,
-    errorInfo
+}
+
+export function ReportFatalError(error: Error, context: ErrorContextTypes, payload: any = null) {
+  const finalPayload = GetErrorPayload(context, payload)
+  trackEvent('error_generic', {
```

Comment:
Is the event `error_generic` fatal enough?

Reply:
ofc not, didn't want to change any tracking event name to not break any existing queries. But I can change it to `error_critical`. In fact I can also change the method name to `ReportCriticalError` to encourage ubiquity. wdyt?
cc: @sicarul

Reply:
I'll defer to @sicarul to decide, you already have my approve